### PR TITLE
Setup playbooks

### PR DESCRIPTION
**Pull request by Timdnb**
Changed the structure and made the playbook that installs Docker and Minikube, I think this structure makes sense, but since I don't really understand yet how to setup the cluster, it might change later, we'll see

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -16,42 +16,65 @@ File.open("inventory.ini", "w") do |file|
...
       worker.vm.provider "virtualbox" do |vb|
         vb.memory = 6144
         vb.cpus = 2
       end
+      
+      # setup worker node with ansible playbook
+      worker.vm.provision "ansible" do |ansible|
+        ansible.compatibility_mode = "2.0"
+        ansible.inventory_path = "inventory.ini"
+        ansible.playbook = "playbook/setup_general.yml"
```
Comment (SagaRut) [inline Vagrantfile:70]: this should be playbooks/ not playbook/

File: Vagrantfile [hunk 2]
```diff
@@ -16,42 +16,65 @@ File.open("inventory.ini", "w") do |file|
...
+      end
 
+      # join worker node in cluster
       worker.vm.provision "ansible" do |ansible|
         ansible.compatibility_mode = "2.0"
         ansible.inventory_path = "inventory.ini"
-        ansible.playbook = "playbook_node.yml"
+        ansible.playbook = "playbook/node.yml"
```
Comment (SagaRut) [inline Vagrantfile:77]: Also this, this should be playbooks/ not playbook/

File: Vagrantfile [hunk 3]
```diff
@@ -16,42 +16,65 @@ File.open("inventory.ini", "w") do |file|
...
 Vagrant.configure("2") do |config|
-  # setup control node
+  # --- Control Node ---
   config.vm.define "controller" do |control|
-    control.vm.box = "bento/ubuntu-24.04"
-    control.vm.box_version = "202404.26.0"
+    control.vm.box = "bento/ubuntu-22.04"
+    control.vm.box_version = "202404.23.0"
     control.vm.hostname = "controller"
     config.vm.network "private_network", ip: "192.168.56.10"
```
Comment (SagaRut) [inline Vagrantfile:24]: Is there a reason this is config.vm.network not control.vm.network?

**Review by SagaRut — CHANGES_REQUESTED** [review]
Also the playbook to set up the cluster for me fails in the minikube start step, sent a message on whatsapp.

File: Vagrantfile (hunk 3 above)
Comment (Timdnb) [reply Vagrantfile:24]: oh huh no, probably made a mistake when updating the code

**Review by SagaRut — APPROVED** [review]

**Pull request merged**